File: 92-reverse-linked-list-ii/reverse-linked-list-ii.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseBetween(struct ListNode* head, int left, int right) {
    if(!head || left==right) return head;
  struct ListNode dummy={0,head};
  struct ListNode*prev=&dummy;
  for(int i=1;i<left;i++){
    prev=prev->next;
  }  
  struct ListNode*start=prev->next;
  struct ListNode*then=start->next;
  for(int i=0;i<right-left;i++){
    start->next=then->next;
    then->next=prev->next;
    prev->next=then;
    then=start->next;
  }
  return dummy.next;
}

File: 92-reverse-linked-list-ii/reverse-linked-list-ii.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseBetween(self, head, left, right):
        dummy=ListNode(0)
        dummy.next=head
        prevm=dummy
        for _ in range(left-1):
            prevm=prevm.next
        prev=None
        curr=prevm.next
        for _ in range(right-left+1):
            nxt=curr.next
            curr.next=prev
            prev=curr
            curr=nxt
        prevm.next.next=curr
        prevm.next=prev
        return dummy.next    

